openlane/scripts/yosys/synthesize.tcl

```diff
@@ -248,120 +98,73 @@ if { $::env(SYNTH_ELABORATE_ONLY) } {
     exit 0
 }
 
-# Infer tri-state buffers.
+# Add various maps
 set tbuf_map false
+set fa_map false
 if { [info exists ::env(SYNTH_TRISTATE_MAP)] } {
     set tbuf_map true
     tribuf
 }
-
-# Handle technology mapping of RCS/CSA adders
+set adder_type $::env(SYNTH_ADDER_TYPE)
+if { !($adder_type in [list "YOSYS" "FA" "RCA" "CSA"]) } {
+    log -stderr "\[ERROR] Misformatted SYNTH_ADDER_TYPE (\"$::env(SYNTH_ADDER_TYPE)\")."
+    log -stderr "\[ERROR] Correct format is \"YOSYS|FA|RCA|CSA\"."
+    exit 1
+}
 if { $adder_type == "RCA"} {
     if { [info exists ::env(SYNTH_RCA_MAP)] } {
-        log "\[INFO] Applying ripple carry adder mapping from '$::env(RIPPLE_CARRY_ADDER_MAP)'…"
+        log "\[INFO] Applying ripple carry adder mapping from '$::env(RIPPLE_CARRY_ADDER_MAP)'..."
```

Comment:
why?

Reply:
Yosys appears to have a problem with Unicode